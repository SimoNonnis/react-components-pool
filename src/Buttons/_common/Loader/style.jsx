import styled from 'styled-components';
import Spinner from '../../../Icons/Spinner';

export const SpinnerContainer = styled.div`
  position: relative;
`;

export const SpinnerStyled = styled(Spinner)`
  position: absolute;
  margin: 0 auto;
  left: 0;
  right: 0;
  top: -3px;
  ${({ size }) => size === 'large' && 'top: -5px'};
  ${({ size }) => size === 'small' && 'top: 0'};
`;
