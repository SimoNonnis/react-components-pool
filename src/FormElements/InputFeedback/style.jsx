import styled from 'styled-components';

export const ErrorStyled = styled.div`
  color: ${({ theme }) => theme.color.red};
  font-size: ${({ theme }) => theme.base.fontSize};
  margin-top: 7px;
  min-height: 17px;
`;
