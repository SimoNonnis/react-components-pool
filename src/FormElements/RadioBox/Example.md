Multiple RadioBoxes with default props

```jsx
initialState = { checked1: false, checked2: true, checked3: false, checked4: false };
<div>
  <RadioBox
    id="banana"
    label="Banana"
    name="fruits"
    checked={state.checked1}
    onChange={() => setState({ checked1: !state.checked1 })}
  />
  <RadioBox
    id="orange"
    label="Orange"
    name="fruits"
    checked={state.checked2}
    onChange={() => setState({ checked2: !state.checked2 })}
  />
  <RadioBox
    id="avocado"
    label="Avocado"
    name="fruits"
    checked={state.checked3}
    onChange={() => setState({ checked3: !state.checked3 })}
  />
  <RadioBox
    id="avocado"
    label="Avocado (disabled)"
    name="fruits"
    disabled
    checked={state.checked4}
    onChange={() => setState({ checked3: !state.checked4 })}
  />
</div>;
```

```jsx
initialState = { checked: false };
<div>
  <RadioBox
    id="longLabel1"
    label="I am a very long label that will wrap to the next line! I am a very long label that will wrap to the next line! I am a very long label that will wrap to the next line!"
    name="longLabel"
    checked={state.checked}
    onChange={() => setState({ checked: !state.checked })}
  />
</div>;
```

```jsx
initialState = { checked: false };
<div>
  <RadioBox
    id="labelWithHtml"
    label={<i>Label with Html</i>}
    name="labelWithHtml"
    checked={state.checked}
    onChange={() => setState({ checked: !state.checked })}
  />
</div>;
```
