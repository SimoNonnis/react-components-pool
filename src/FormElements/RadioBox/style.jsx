import styled from 'styled-components';

export const Container = styled.span`
  display: inline-block;
  position: relative;

  & + & {
    margin-left: 1em;
  }
`;
Container.displayName = 'Container';

export const Radio = styled.input`
  opacity: 0;
  position: absolute;
  right: 0;

  &:checked + label:after {
    opacity: 1;
    transform: scale(1);
  }
`;
Radio.displayName = 'Radio';

export const Label = styled.label`
  cursor: pointer;
  font-family: ${({ theme }) => theme.base.fontFamily};
  color: ${({ theme }) => theme.color.veryDarkBlue};
  ${({ theme, error }) => error && `color: ${theme.color.red}`};
  ${({ theme, disabled }) => disabled && `color: ${theme.color.lightGrey1}`};
  font-size: 16px;
  line-height: 24px;
  position: relative;
  user-select: none;
  padding-left: 30px;
  display: inline-block;
  font-weight: 600;

  &:before {
    box-sizing: content-box;
    background-color: ${({ theme }) => theme.color.white};
    border-radius: 100%;
    border: 1px solid;
    border-color: ${({ theme: { color }, disabled }) =>
      disabled ? color.lightGrey1 : color.stoneGrey};
    box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.1);
    content: '';
    height: 20px;
    left: 0;
    position: absolute;
    top: 0;
    width: 20px;
  }

  ${({ theme, disabled }) =>
    disabled ? `&:hover { &:before { background-color: ${theme.color.lightGrey3}; } }` : null};
  &:after {
    box-sizing: content-box;
    background-color: ${({ theme }) => theme.color.mediumBlue};
    border-radius: 100%;
    content: '';
    opacity: 0;
    position: absolute;
    transform: scale(0);
    transition: all 0.2s;
    height: 12px;
    width: 12px;
    top: 5px;
    left: 5px;
  }
`;
Label.displayName = 'Label';
