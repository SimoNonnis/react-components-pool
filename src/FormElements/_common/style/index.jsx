import styled, { css } from 'styled-components';

const basicStyle = css`
  box-shadow: none;
  box-sizing: border-box;
  display: block;
  font-family: ${({ theme }) => theme.base.fontFamily};
  font-size: 16px;
  font-weight: inherit;
  height: auto;
  line-height: 24px;
  margin: 0;
  width: 100%;
  &:focus {
    outline: none;
    border-color: ${({ theme }) => theme.color.mediumBlue};
  }
`;

const borderStyle = css`
  border-radius: 2px;
  border: 1px solid;
`;

const defaultColor = css`
  ${borderStyle};
  border-color: ${({ theme }) => theme.color.stoneGrey};
  color: ${({ theme }) => theme.color.veryDarkBlue};
  caret-color: ${({ theme }) => theme.color.veryDarkBlue};
  &:focus {
    border-color: ${({ theme }) => theme.color.mediumBlue};
  }
  &::-webkit-datetime-edit-year-field,
  &::-webkit-datetime-edit-day-field,
  &::-webkit-datetime-edit-month-field {
    color: ${({ theme }) => theme.color.veryDarkBlue};
    font-weight: inherit;
  }
`;

const errorColor = css`
  ${borderStyle};
  border-color: ${({ theme }) => theme.color.red};
  &:focus {
    border-color: ${({ theme }) => theme.color.red};
  }
`;

const disabledColor = css`
  color: ${({ theme }) => theme.color.lightGrey2};
  border-color: ${({ theme }) => theme.color.lightGrey2};
  caret-color: ${({ theme }) => theme.color.lightGrey2};
  &:focus {
    border-color: ${({ theme }) => theme.color.lightGrey2};
  }
  &::-webkit-datetime-edit-year-field,
  &::-webkit-datetime-edit-day-field,
  &::-webkit-datetime-edit-month-field {
    color: ${({ theme }) => theme.color.lightGrey2};
    font-weight: inherit;
  }
`;

export const InputStyled = styled.input`
  ${basicStyle};
  ${defaultColor};
  ${({ error }) => error && errorColor};
  ${({ disabled }) => disabled && disabledColor};
  padding: 12px 15px;
`;

export const SelectStyled = styled.select`
  ${basicStyle};
  color: ${({ theme, disabled }) => (disabled ? theme.color.lightGrey2 : theme.color.veryDarkBlue)};
  padding: 13px 50px 13px 15px;
  -ms-appearance: none !important;
  appearance: none !important;
  background-image: none !important;
  background: transparent !important;
  border-radius: 0 !important;
  border: 0 !important;
  box-shadow: none !important;
  outline: 0 !important;
`;

export const SelectWrap = styled.div`
  ${defaultColor};
  ${({ error }) => error && errorColor};
  ${({ disabled }) => disabled && disabledColor};
  position: relative;
  &:after {
    border-left: 6px solid transparent;
    border-right: 6px solid transparent;
    border-top: 8px solid
      ${({ theme, disabled }) => (disabled ? theme.color.lightGrey2 : theme.color.veryDarkBlue)};
    content: '';
    height: 0;
    pointer-events: none;
    position: absolute;
    right: 20px;
    top: 50%;
    transform: translateY(-50%);
    transition: all 0.2s cubic-bezier(0.25, 0.46, 0.45, 0.94);
    width: 0;
    z-index: 2;
  }
`;

export const TextareaStyled = styled.textarea`
  ${basicStyle};
  ${defaultColor};
  ${({ error }) => error && errorColor};
  ${({ disabled }) => disabled && disabledColor};
  padding: 15px;
  resize: ${({ disabled }) => (disabled ? 'none' : 'vertical')};
`;
