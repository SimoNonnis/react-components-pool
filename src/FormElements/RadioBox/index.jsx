import React from 'react';
import PropTypes from 'prop-types';

import { Container, Radio, Label } from './style';

const propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.oneOfType([PropTypes.func, PropTypes.node]).isRequired,
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
  checked: PropTypes.bool,
  error: PropTypes.string,
  disabled: PropTypes.bool,
};
const defaultProps = {
  className: null,
  checked: false,
  error: null,
  disabled: false,
};

const RadioBox = ({ className, id, name, label, error, checked, disabled, ...props }) => {
  const labelComponent = typeof label === 'function' ? label() : label;

  return (
    <Container className={className}>
      <Radio disabled={disabled} type="radio" id={id} name={name} checked={checked} {...props} />
      <Label disabled={disabled} htmlFor={id} error={!!error}>
        {labelComponent}
      </Label>
    </Container>
  );
};

RadioBox.propTypes = propTypes;
RadioBox.defaultProps = defaultProps;

export default RadioBox;
