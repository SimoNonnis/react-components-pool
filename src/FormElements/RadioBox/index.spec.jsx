/* eslint-env jest */
import React from 'react';

import 'jest-styled-components';
import styled from 'styled-components';

import theme from 'themes';
import RadioBox from './';
import { Radio } from './style';

export const StyledLabel = styled.div`
  color: coral;
`;

describe('Test <RadioBox />', () => {
  it('should render RadioBox correctly with only required props', () => {
    const wrapper = shallow(<RadioBox id="ok" label="Ok" name="cool" theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render RadioBox correctly if checked true', () => {
    const wrapper = shallow(<RadioBox id="no" label="No" name="cool" checked theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChange when RadioBox is clicked', () => {
    const handleChangeMock = jest.fn();
    const wrapper = shallow(
      <RadioBox
        id="yes"
        label="Yes"
        name="choice"
        checked
        onChange={handleChangeMock}
        theme={theme}
      />
    );
    wrapper.find(Radio).simulate('change');
    expect(handleChangeMock).toHaveBeenCalled();
  });

  it('should render RadioBox correctly with label as a function', () => {
    const wrapper = shallow(
      <RadioBox id="yes" label={() => <span>Label component</span>} name="choice" theme={theme} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render RadioBox correctly component with label as a styled component', () => {
    const testLabel = <StyledLabel>Label</StyledLabel>;
    const wrapper = shallow(
      <RadioBox id="yes" label={testLabel} name="choice" error="testError" theme={theme} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render RadioBox correctly component with label containing html tags', () => {
    const wrapper = shallow(
      <RadioBox id="yes" label={<span>Label</span>} name="choice" error="testError" theme={theme} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleBlur when RadioBox is clicked', () => {
    const handleChangeMock = jest.fn();
    const handleBlurMock = jest.fn();

    const wrapper = shallow(
      <RadioBox
        id="yes"
        label="Yes"
        name="choice"
        checked
        onChange={handleChangeMock}
        onBlur={handleBlurMock}
        theme={theme}
      />
    );
    wrapper.find(Radio).simulate('blur');
    expect(handleBlurMock).toHaveBeenCalled();
  });
});
