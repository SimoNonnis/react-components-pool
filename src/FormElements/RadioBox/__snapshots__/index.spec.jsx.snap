// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <RadioBox /> should render RadioBox correctly component with label as a styled component 1`] = `
<Container
  className={null}
>
  <Radio
    checked={false}
    disabled={false}
    id="yes"
    name="choice"
    theme={
      Object {
        "base": Object {
          "fontFamily": "\\"proxima-nova\\", \\"Open Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1.45,
        },
        "breakpoints": Array [
          504,
          671,
          867,
          1050,
          1200,
        ],
        "color": Object {
          "black": "#000000",
          "coral": "#F4364C",
          "darkGrey": "#77838d",
          "green": "#35A856",
          "lightBlue1": "#daf3ff",
          "lightBlue2": "#bfe8f8",
          "lightGreen": "#B4E2C2",
          "lightGrey1": "#d6d6d6",
          "lightGrey2": "#e6e6e6",
          "lightGrey3": "#f6f6f6",
          "lightGrey4": "#c8cccf",
          "lightOrange": "#ffeaca",
          "lightPink": "#f8d7e0",
          "mediumBlue": "#0099cf",
          "mediumGrey": "#979797",
          "orange": "#ffad2f",
          "pink": "#e46185",
          "red": "#f80000",
          "red15": "#fedede",
          "stoneGrey": "#a6adb4",
          "teal": "#06A693",
          "veryDarkBlue": "#1D3141",
          "veryDarkBlue20": "#C8CCCF",
          "veryDarkBlue40": "#9DA5AB",
          "veryDarkBlue60": "#717D87",
          "veryDarkBlue7": "#EEF0F3",
          "veryDarkBlue80": "#4A5A67",
          "white": "#ffffff",
        },
        "maxWidth": "1200px",
        "space": Array [
          5,
          10,
          15,
          20,
          30,
        ],
        "spacing": Object {
          "lg": "28px",
          "md": "18px",
          "sm": "10px",
          "xl": "36px",
          "xs": "5px",
        },
        "typography": Object {
          "caption": Object {
            "desktop": Object {
              "fontSize": "1rem",
              "lineHeight": "1.571rem",
            },
            "phone": Object {
              "fontSize": "1rem",
              "fontWeight": 400,
              "lineHeight": "1.571rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1rem",
              "lineHeight": "1.571rem",
            },
          },
          "h1": Object {
            "desktop": Object {
              "fontSize": "2.286rem",
              "lineHeight": "2.571rem",
            },
            "phone": Object {
              "fontSize": "1.714rem",
              "fontWeight": 600,
              "lineHeight": "2rem",
              "margin": "0 0 36px 0",
            },
            "tablet": Object {
              "fontSize": "2rem",
              "lineHeight": "2.429rem",
            },
          },
          "h2": Object {
            "desktop": Object {
              "fontSize": "1.571rem",
              "lineHeight": "1.857rem",
            },
            "phone": Object {
              "fontSize": "1.429rem",
              "fontWeight": 600,
              "lineHeight": "1.714rem",
              "margin": "0 0 18px 0",
            },
            "tablet": Object {
              "fontSize": "1.429rem",
              "lineHeight": "1.714rem",
            },
          },
          "h3": Object {
            "desktop": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
            "phone": Object {
              "fontSize": "1.143rem",
              "fontWeight": 700,
              "lineHeight": "1.714rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
          },
          "intro": Object {
            "desktop": Object {
              "fontSize": "1.357rem",
              "lineHeight": "1.929rem",
            },
            "phone": Object {
              "fontSize": "1.214rem",
              "fontWeight": 300,
              "lineHeight": "1.714rem",
              "margin": "0 0 36px 0",
            },
            "tablet": Object {
              "fontSize": "1.214rem",
              "lineHeight": "1.714rem",
            },
          },
          "p": Object {
            "desktop": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
            "phone": Object {
              "fontSize": "1.143rem",
              "fontWeight": 400,
              "lineHeight": "1.714rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
          },
        },
      }
    }
    type="radio"
  />
  <Label
    disabled={false}
    error={true}
    htmlFor="yes"
  >
    <styled.div>
      Label
    </styled.div>
  </Label>
</Container>
`;

exports[`Test <RadioBox /> should render RadioBox correctly component with label containing html tags 1`] = `
<Container
  className={null}
>
  <Radio
    checked={false}
    disabled={false}
    id="yes"
    name="choice"
    theme={
      Object {
        "base": Object {
          "fontFamily": "\\"proxima-nova\\", \\"Open Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1.45,
        },
        "breakpoints": Array [
          504,
          671,
          867,
          1050,
          1200,
        ],
        "color": Object {
          "black": "#000000",
          "coral": "#F4364C",
          "darkGrey": "#77838d",
          "green": "#35A856",
          "lightBlue1": "#daf3ff",
          "lightBlue2": "#bfe8f8",
          "lightGreen": "#B4E2C2",
          "lightGrey1": "#d6d6d6",
          "lightGrey2": "#e6e6e6",
          "lightGrey3": "#f6f6f6",
          "lightGrey4": "#c8cccf",
          "lightOrange": "#ffeaca",
          "lightPink": "#f8d7e0",
          "mediumBlue": "#0099cf",
          "mediumGrey": "#979797",
          "orange": "#ffad2f",
          "pink": "#e46185",
          "red": "#f80000",
          "red15": "#fedede",
          "stoneGrey": "#a6adb4",
          "teal": "#06A693",
          "veryDarkBlue": "#1D3141",
          "veryDarkBlue20": "#C8CCCF",
          "veryDarkBlue40": "#9DA5AB",
          "veryDarkBlue60": "#717D87",
          "veryDarkBlue7": "#EEF0F3",
          "veryDarkBlue80": "#4A5A67",
          "white": "#ffffff",
        },
        "maxWidth": "1200px",
        "space": Array [
          5,
          10,
          15,
          20,
          30,
        ],
        "spacing": Object {
          "lg": "28px",
          "md": "18px",
          "sm": "10px",
          "xl": "36px",
          "xs": "5px",
        },
        "typography": Object {
          "caption": Object {
            "desktop": Object {
              "fontSize": "1rem",
              "lineHeight": "1.571rem",
            },
            "phone": Object {
              "fontSize": "1rem",
              "fontWeight": 400,
              "lineHeight": "1.571rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1rem",
              "lineHeight": "1.571rem",
            },
          },
          "h1": Object {
            "desktop": Object {
              "fontSize": "2.286rem",
              "lineHeight": "2.571rem",
            },
            "phone": Object {
              "fontSize": "1.714rem",
              "fontWeight": 600,
              "lineHeight": "2rem",
              "margin": "0 0 36px 0",
            },
            "tablet": Object {
              "fontSize": "2rem",
              "lineHeight": "2.429rem",
            },
          },
          "h2": Object {
            "desktop": Object {
              "fontSize": "1.571rem",
              "lineHeight": "1.857rem",
            },
            "phone": Object {
              "fontSize": "1.429rem",
              "fontWeight": 600,
              "lineHeight": "1.714rem",
              "margin": "0 0 18px 0",
            },
            "tablet": Object {
              "fontSize": "1.429rem",
              "lineHeight": "1.714rem",
            },
          },
          "h3": Object {
            "desktop": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
            "phone": Object {
              "fontSize": "1.143rem",
              "fontWeight": 700,
              "lineHeight": "1.714rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
          },
          "intro": Object {
            "desktop": Object {
              "fontSize": "1.357rem",
              "lineHeight": "1.929rem",
            },
            "phone": Object {
              "fontSize": "1.214rem",
              "fontWeight": 300,
              "lineHeight": "1.714rem",
              "margin": "0 0 36px 0",
            },
            "tablet": Object {
              "fontSize": "1.214rem",
              "lineHeight": "1.714rem",
            },
          },
          "p": Object {
            "desktop": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
            "phone": Object {
              "fontSize": "1.143rem",
              "fontWeight": 400,
              "lineHeight": "1.714rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
          },
        },
      }
    }
    type="radio"
  />
  <Label
    disabled={false}
    error={true}
    htmlFor="yes"
  >
    <span>
      Label
    </span>
  </Label>
</Container>
`;

exports[`Test <RadioBox /> should render RadioBox correctly if checked true 1`] = `
<Container
  className={null}
>
  <Radio
    checked={true}
    disabled={false}
    id="no"
    name="cool"
    theme={
      Object {
        "base": Object {
          "fontFamily": "\\"proxima-nova\\", \\"Open Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1.45,
        },
        "breakpoints": Array [
          504,
          671,
          867,
          1050,
          1200,
        ],
        "color": Object {
          "black": "#000000",
          "coral": "#F4364C",
          "darkGrey": "#77838d",
          "green": "#35A856",
          "lightBlue1": "#daf3ff",
          "lightBlue2": "#bfe8f8",
          "lightGreen": "#B4E2C2",
          "lightGrey1": "#d6d6d6",
          "lightGrey2": "#e6e6e6",
          "lightGrey3": "#f6f6f6",
          "lightGrey4": "#c8cccf",
          "lightOrange": "#ffeaca",
          "lightPink": "#f8d7e0",
          "mediumBlue": "#0099cf",
          "mediumGrey": "#979797",
          "orange": "#ffad2f",
          "pink": "#e46185",
          "red": "#f80000",
          "red15": "#fedede",
          "stoneGrey": "#a6adb4",
          "teal": "#06A693",
          "veryDarkBlue": "#1D3141",
          "veryDarkBlue20": "#C8CCCF",
          "veryDarkBlue40": "#9DA5AB",
          "veryDarkBlue60": "#717D87",
          "veryDarkBlue7": "#EEF0F3",
          "veryDarkBlue80": "#4A5A67",
          "white": "#ffffff",
        },
        "maxWidth": "1200px",
        "space": Array [
          5,
          10,
          15,
          20,
          30,
        ],
        "spacing": Object {
          "lg": "28px",
          "md": "18px",
          "sm": "10px",
          "xl": "36px",
          "xs": "5px",
        },
        "typography": Object {
          "caption": Object {
            "desktop": Object {
              "fontSize": "1rem",
              "lineHeight": "1.571rem",
            },
            "phone": Object {
              "fontSize": "1rem",
              "fontWeight": 400,
              "lineHeight": "1.571rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1rem",
              "lineHeight": "1.571rem",
            },
          },
          "h1": Object {
            "desktop": Object {
              "fontSize": "2.286rem",
              "lineHeight": "2.571rem",
            },
            "phone": Object {
              "fontSize": "1.714rem",
              "fontWeight": 600,
              "lineHeight": "2rem",
              "margin": "0 0 36px 0",
            },
            "tablet": Object {
              "fontSize": "2rem",
              "lineHeight": "2.429rem",
            },
          },
          "h2": Object {
            "desktop": Object {
              "fontSize": "1.571rem",
              "lineHeight": "1.857rem",
            },
            "phone": Object {
              "fontSize": "1.429rem",
              "fontWeight": 600,
              "lineHeight": "1.714rem",
              "margin": "0 0 18px 0",
            },
            "tablet": Object {
              "fontSize": "1.429rem",
              "lineHeight": "1.714rem",
            },
          },
          "h3": Object {
            "desktop": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
            "phone": Object {
              "fontSize": "1.143rem",
              "fontWeight": 700,
              "lineHeight": "1.714rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
          },
          "intro": Object {
            "desktop": Object {
              "fontSize": "1.357rem",
              "lineHeight": "1.929rem",
            },
            "phone": Object {
              "fontSize": "1.214rem",
              "fontWeight": 300,
              "lineHeight": "1.714rem",
              "margin": "0 0 36px 0",
            },
            "tablet": Object {
              "fontSize": "1.214rem",
              "lineHeight": "1.714rem",
            },
          },
          "p": Object {
            "desktop": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
            "phone": Object {
              "fontSize": "1.143rem",
              "fontWeight": 400,
              "lineHeight": "1.714rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
          },
        },
      }
    }
    type="radio"
  />
  <Label
    disabled={false}
    error={false}
    htmlFor="no"
  >
    No
  </Label>
</Container>
`;

exports[`Test <RadioBox /> should render RadioBox correctly with label as a function 1`] = `
<Container
  className={null}
>
  <Radio
    checked={false}
    disabled={false}
    id="yes"
    name="choice"
    theme={
      Object {
        "base": Object {
          "fontFamily": "\\"proxima-nova\\", \\"Open Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1.45,
        },
        "breakpoints": Array [
          504,
          671,
          867,
          1050,
          1200,
        ],
        "color": Object {
          "black": "#000000",
          "coral": "#F4364C",
          "darkGrey": "#77838d",
          "green": "#35A856",
          "lightBlue1": "#daf3ff",
          "lightBlue2": "#bfe8f8",
          "lightGreen": "#B4E2C2",
          "lightGrey1": "#d6d6d6",
          "lightGrey2": "#e6e6e6",
          "lightGrey3": "#f6f6f6",
          "lightGrey4": "#c8cccf",
          "lightOrange": "#ffeaca",
          "lightPink": "#f8d7e0",
          "mediumBlue": "#0099cf",
          "mediumGrey": "#979797",
          "orange": "#ffad2f",
          "pink": "#e46185",
          "red": "#f80000",
          "red15": "#fedede",
          "stoneGrey": "#a6adb4",
          "teal": "#06A693",
          "veryDarkBlue": "#1D3141",
          "veryDarkBlue20": "#C8CCCF",
          "veryDarkBlue40": "#9DA5AB",
          "veryDarkBlue60": "#717D87",
          "veryDarkBlue7": "#EEF0F3",
          "veryDarkBlue80": "#4A5A67",
          "white": "#ffffff",
        },
        "maxWidth": "1200px",
        "space": Array [
          5,
          10,
          15,
          20,
          30,
        ],
        "spacing": Object {
          "lg": "28px",
          "md": "18px",
          "sm": "10px",
          "xl": "36px",
          "xs": "5px",
        },
        "typography": Object {
          "caption": Object {
            "desktop": Object {
              "fontSize": "1rem",
              "lineHeight": "1.571rem",
            },
            "phone": Object {
              "fontSize": "1rem",
              "fontWeight": 400,
              "lineHeight": "1.571rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1rem",
              "lineHeight": "1.571rem",
            },
          },
          "h1": Object {
            "desktop": Object {
              "fontSize": "2.286rem",
              "lineHeight": "2.571rem",
            },
            "phone": Object {
              "fontSize": "1.714rem",
              "fontWeight": 600,
              "lineHeight": "2rem",
              "margin": "0 0 36px 0",
            },
            "tablet": Object {
              "fontSize": "2rem",
              "lineHeight": "2.429rem",
            },
          },
          "h2": Object {
            "desktop": Object {
              "fontSize": "1.571rem",
              "lineHeight": "1.857rem",
            },
            "phone": Object {
              "fontSize": "1.429rem",
              "fontWeight": 600,
              "lineHeight": "1.714rem",
              "margin": "0 0 18px 0",
            },
            "tablet": Object {
              "fontSize": "1.429rem",
              "lineHeight": "1.714rem",
            },
          },
          "h3": Object {
            "desktop": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
            "phone": Object {
              "fontSize": "1.143rem",
              "fontWeight": 700,
              "lineHeight": "1.714rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
          },
          "intro": Object {
            "desktop": Object {
              "fontSize": "1.357rem",
              "lineHeight": "1.929rem",
            },
            "phone": Object {
              "fontSize": "1.214rem",
              "fontWeight": 300,
              "lineHeight": "1.714rem",
              "margin": "0 0 36px 0",
            },
            "tablet": Object {
              "fontSize": "1.214rem",
              "lineHeight": "1.714rem",
            },
          },
          "p": Object {
            "desktop": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
            "phone": Object {
              "fontSize": "1.143rem",
              "fontWeight": 400,
              "lineHeight": "1.714rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
          },
        },
      }
    }
    type="radio"
  />
  <Label
    disabled={false}
    error={false}
    htmlFor="yes"
  >
    <span>
      Label component
    </span>
  </Label>
</Container>
`;

exports[`Test <RadioBox /> should render RadioBox correctly with only required props 1`] = `
<Container
  className={null}
>
  <Radio
    checked={false}
    disabled={false}
    id="ok"
    name="cool"
    theme={
      Object {
        "base": Object {
          "fontFamily": "\\"proxima-nova\\", \\"Open Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1.45,
        },
        "breakpoints": Array [
          504,
          671,
          867,
          1050,
          1200,
        ],
        "color": Object {
          "black": "#000000",
          "coral": "#F4364C",
          "darkGrey": "#77838d",
          "green": "#35A856",
          "lightBlue1": "#daf3ff",
          "lightBlue2": "#bfe8f8",
          "lightGreen": "#B4E2C2",
          "lightGrey1": "#d6d6d6",
          "lightGrey2": "#e6e6e6",
          "lightGrey3": "#f6f6f6",
          "lightGrey4": "#c8cccf",
          "lightOrange": "#ffeaca",
          "lightPink": "#f8d7e0",
          "mediumBlue": "#0099cf",
          "mediumGrey": "#979797",
          "orange": "#ffad2f",
          "pink": "#e46185",
          "red": "#f80000",
          "red15": "#fedede",
          "stoneGrey": "#a6adb4",
          "teal": "#06A693",
          "veryDarkBlue": "#1D3141",
          "veryDarkBlue20": "#C8CCCF",
          "veryDarkBlue40": "#9DA5AB",
          "veryDarkBlue60": "#717D87",
          "veryDarkBlue7": "#EEF0F3",
          "veryDarkBlue80": "#4A5A67",
          "white": "#ffffff",
        },
        "maxWidth": "1200px",
        "space": Array [
          5,
          10,
          15,
          20,
          30,
        ],
        "spacing": Object {
          "lg": "28px",
          "md": "18px",
          "sm": "10px",
          "xl": "36px",
          "xs": "5px",
        },
        "typography": Object {
          "caption": Object {
            "desktop": Object {
              "fontSize": "1rem",
              "lineHeight": "1.571rem",
            },
            "phone": Object {
              "fontSize": "1rem",
              "fontWeight": 400,
              "lineHeight": "1.571rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1rem",
              "lineHeight": "1.571rem",
            },
          },
          "h1": Object {
            "desktop": Object {
              "fontSize": "2.286rem",
              "lineHeight": "2.571rem",
            },
            "phone": Object {
              "fontSize": "1.714rem",
              "fontWeight": 600,
              "lineHeight": "2rem",
              "margin": "0 0 36px 0",
            },
            "tablet": Object {
              "fontSize": "2rem",
              "lineHeight": "2.429rem",
            },
          },
          "h2": Object {
            "desktop": Object {
              "fontSize": "1.571rem",
              "lineHeight": "1.857rem",
            },
            "phone": Object {
              "fontSize": "1.429rem",
              "fontWeight": 600,
              "lineHeight": "1.714rem",
              "margin": "0 0 18px 0",
            },
            "tablet": Object {
              "fontSize": "1.429rem",
              "lineHeight": "1.714rem",
            },
          },
          "h3": Object {
            "desktop": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
            "phone": Object {
              "fontSize": "1.143rem",
              "fontWeight": 700,
              "lineHeight": "1.714rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
          },
          "intro": Object {
            "desktop": Object {
              "fontSize": "1.357rem",
              "lineHeight": "1.929rem",
            },
            "phone": Object {
              "fontSize": "1.214rem",
              "fontWeight": 300,
              "lineHeight": "1.714rem",
              "margin": "0 0 36px 0",
            },
            "tablet": Object {
              "fontSize": "1.214rem",
              "lineHeight": "1.714rem",
            },
          },
          "p": Object {
            "desktop": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
            "phone": Object {
              "fontSize": "1.143rem",
              "fontWeight": 400,
              "lineHeight": "1.714rem",
              "margin": "0 0 10px 0",
            },
            "tablet": Object {
              "fontSize": "1.143rem",
              "lineHeight": "1.714rem",
            },
          },
        },
      }
    }
    type="radio"
  />
  <Label
    disabled={false}
    error={false}
    htmlFor="ok"
  >
    Ok
  </Label>
</Container>
`;
